Sources/ClientRuntime/Config/DefaultClientConfiguration.swift

```diff
@@ -29,5 +29,8 @@ public protocol DefaultClientConfiguration: ClientConfiguration {
     /// A token generator to ensure idempotency of requests.
     var idempotencyTokenGenerator: IdempotencyTokenGenerator { get set }
 
-    /// TODO(plugins): Add Checksum, Traceprobes, etc.
+    /// A checksum algorithm used for requests.
+    var checksumAlgorithm: ChecksumAlgorithm? { get set }
```

Comment:
Just double checking, do we need to support multiple checksum algorithms on a client configuration at once? e.g. [like smithy-typescript](https://github.com/smithy-lang/smithy-typescript/blob/e03a10ac1f34ac52e0b03509c13499e459bde3bc/packages/smithy-client/src/extensions/checksum.ts#L9-L15)

If not, then LGTM

Reply:
Question: Should the plugin use ChecksumAlgorithm enum (all possible supported checksum algorithms) or specifically the ones the particular client supports by looking in XClientTypes?